Fixed topic 'Building This Project'

oldfartdeveloper opened the pull request:
Fixes: #575

1. Rebuilt this capability
2. Cleaned up the install/build/test reporting.
3. In script `spago--build-and-test.sh`, changed compare operator `==` to `=` and made is a `sh` script instead of a `bash` script to upgrade to fix a MacOS **Catalina** syntax incompatibility.

The above changes were made so that continuous integration facility's functionality was left unchanged.  The edits look more extensive than they really are in that I had to move large chunks of the shell code from one script to another.

oldfartdeveloper committed:
Fix .ci scripts according to Jordan's recommendation

JordanMartinez commented:
Sorry for taking a while to get to this. Thanks for doing this!

JordanMartinez commented:
I think CI is failing because `.github/workflows/ci.yml` never makes `.ci/spago--build-and-test-non-ci.sh` executable via `chmod +x .ci/spago--build-and-test-non-ci.sh`. But if that's the case, can we drop `.ci/spago--build-and-test.sh` and just call `exit "`source .ci/spago--build-and-test-non-ci.sh`"` directly as a step in the GH Action? And if so, then could we drop the `-non-ci` part of that file's name?

oldfartdeveloper commented:
Have suggested Jordan fixes.

oldfartdeveloper commented:
I've submitted PR #577 that has:

1. Incorporated your suggestions.
2. Removed some ambiguity that I found in my text in this PR.
3. Verified that the CI runs successfully on my fork.

Hence, closing this PR so that you can concentrate on #577
